pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -134,7 +137,7 @@ func (p *PlaceholderPodController) Reconcile(ctx context.Context, req ctrl.Reque
 }
 
 func (p *PlaceholderPodController) buildDeployment(dep *appsv1.Deployment, spc *secv1.SecretProviderClass, ing *netv1.Ingress) {
-	labels := map[string]string{"app": spc.Name}
+	labels := util.MergeMaps(map[string]string{"app": spc.Name}, spc.Labels)
```

Comment:
we don't want to merge with the spc labels. We should merge with our top level labels.